types/json.go

```diff
@@ -125,3 +126,42 @@ func UnmarshalJSON(b []byte, v *Value) error {
 	}
 	return nil
 }
+
+func unmarshalExtensionValue[T any](b []byte, extName string, parse func(string) (T, error)) (T, error) {
+	var zeroT T
+	var arg string
```

Comment:
nit: you could combine this into one var declaration

```suggestion
	var (
		zeroT T
		arg string
	)
```

Reply:
I think I prefer the two line version...